### PR TITLE
added a new admeme weapon: foam finger that shoots missiles as requested by boris (now in arcades!)

### PR DESCRIPTION
**Pull request by c420-o**
## About The Pull Request
image
Boris wanted it, I delivered.

image
Coiax wanted it, I DELIVERED

image
**RAVE WANTED IT. I DELIVERED**

## Changelog
:cl:
add: Added foam finger, toy and admeme version 
admin: added foam finger
/:cl:

### CODE REVIEW

**Comment by RaveRadbury** [comment]
https://www.youtube.com/watch?v=xhU6zDzQPVI

BANG! BANG! BANG! BANG! BANG! 
PEW! PEW! PEW! PEW! PEW! PEW!

**Comment by tralezab** [comment]
FUCK YOU AND FUCK YOUR MARKER

**Comment by c420-o** [comment]
If this doesn't get added I will be mildly upset for like an hour

**Comment by Fikou** [comment]
sneed

**Comment by coiax** [comment]
Can you also add a toy version that is just a giant foam finger?

**Comment by c420-o** [comment]
> Can you also add a toy version that is just a giant foam finger?

Sure! I'll add that this afternoon

File: code/modules/projectiles/guns/ballistic/launchers.dm [hunk 1]
```diff
@@ -109,7 +109,25 @@
 		sleep(20)
 		return OXYLOSS
 
-
+/obj/item/gun/ballistic/foamfinger
+	name = "foamfinger"
```
Comment (coiax) [inline code/modules/projectiles/guns/ballistic/launchers.dm:113]: Make sure the admin one has the same `name` as the toy.

Comment (c420-o) [reply code/modules/projectiles/guns/ballistic/launchers.dm]: done

**Comment by coiax** [comment]
I mean, I want to approve this, but that changelog is so confusing. @Mothblocks are you okay being a sneedtainer

**Comment by c420-o** [comment]
> I mean, I want to approve this, but that changelog is so confusing. @Mothblocks are you okay being a sneedtainer

I'll fix it

**Comment by ShibaInuLord** [comment]
sneedtainer

**Comment by RaveRadbury** [comment]
If you could have it copy the "waves around" action from picket signs that'd be neat, but maybe its better saved for a PR down the line.

**Comment by c420-o** [comment]
> If you could have it copy the "waves around" action from picket signs that'd be neat, but maybe its better saved for a PR down the line.

done.

**Comment by cacogen** [comment]
trickle-down features

**Comment by c420-o** [comment]
> trickle-down features

what did he mean by this

**Comment by c420-o** [comment]
It's all set to go

**Comment by cacogen** [comment]
I meant most players won't get to use this

**Comment by c420-o** [comment]
> I meant most players won't get to use this

That's why it's an admeme weapon. The toy version also isn't really meant to be used alot either since toys aren't used at all unless you're trying to grief with the dummy

**Comment by cacogen** [comment]
Okay well it's a weird mechanic where you have to become an admin to unlock this but I'm sure some people are up to the challenge

**Comment by RaveRadbury** [comment]
Yeah the toy should probably get tossed into the arcade lootpool

**Comment by c420-o** [comment]
> Yeah the toy should probably get tossed into the arcade lootpool

IIRC everything in toy.dm is in the arcade lootpool, but I'll double check

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -1536,3 +1536,34 @@
...
+
+/obj/item/toy/foamfinger
+	name = "foam finger"
+	desc = "root for the home team! wait, does this station even have a sports team?"
+	icon = 'icons/obj/guns/projectile.dmi'
+	icon_state = "foamfinger"
+	inhand_icon_state = "foamfinger_inhand"
+	lefthand_file = 'icons/mob/inhands/weapons/guns_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/guns_righthand.dmi'
+	w_class = WEIGHT_CLASS_NORMAL
```
Comment (coiax) [inline code/game/objects/items/toys.dm:1548]: Make sure admin version is also same size.

**Review by coiax — CHANGES_REQUESTED** [review]
Make sure it's in the arcade toy loot pool, and this is good to go.

**Comment by coiax** [comment]
Also, we'll need sign off from a spriter.

File: code/game/objects/items/toys.dm [hunk 3]
```diff
@@ -1536,3 +1536,34 @@
...
+		animate(pixel_x = user.pixel_x - (2 * direction), time = 1, easing = SINE_EASING)
+		animate(pixel_x = user.pixel_x + (2 * direction), time = 1, easing = SINE_EASING)
+		animate(pixel_x = user.pixel_x - (2 * direction), time = 1, easing = SINE_EASING)
+		animate(pixel_x = user.pixel_x + (1 * direction), time = 1, easing = SINE_EASING)
+	else
+		animate(user, pixel_y = user.pixel_y + (1 * direction), time = 1, easing = SINE_EASING)
+		animate(pixel_y = user.pixel_y - (2 * direction), time = 1, easing = SINE_EASING)
+		animate(pixel_y = user.pixel_y + (2 * direction), time = 1, easing = SINE_EASING)
+		animate(pixel_y = user.pixel_y - (2 * direction), time = 1, easing = SINE_EASING)
+		animate(pixel_y = user.pixel_y + (1 * direction), time = 1, easing = SINE_EASING)
```
Comment (coiax) [inline code/game/objects/items/toys.dm:1568]: Actually is this code the same as the sign?

Comment (c420-o) [reply code/game/objects/items/toys.dm:1568]: Yes

File: code/game/objects/items/toys.dm (hunk 2 above)
Comment (c420-o) [reply code/game/objects/items/toys.dm:1548]: Sure

**Review by c420-o — COMMENTED** [review]
gg ez. all requests by all parties have been added.

**Review by c420-o — COMMENTED** [review]
for some reason didn't do the review right the first time

**Review by Twaticus — APPROVED** [review]

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-2). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**